### PR TITLE
Add default_max_records_num to dozer-config to allow configurable default limit

### PR DESCRIPTION
**Pull request by aaryaattrey**

### CODE REVIEW

File: dozer-api/src/grpc/client_server.rs [hunk 1]
```diff
@@ -40,6 +40,7 @@ impl ApiServer {
         &self,
         cache_endpoints: Vec<Arc<CacheEndpoint>>,
         operations_receiver: Option<broadcast::Receiver<Operation>>,
+        limit: usize,
```
Comment (chubei) [inline dozer-api/src/grpc/client_server.rs:43]: Rename to `default_max_num_records `

File: dozer-api/src/grpc/common/service.rs [hunk 2]
```diff
@@ -26,12 +26,14 @@ pub struct CommonService {
     /// For look up endpoint from its name. `key == value.endpoint.name`. Using index map to keep endpoint order.
     pub endpoint_map: IndexMap<String, Arc<CacheEndpoint>>,
     pub event_notifier: Option<tokio::sync::broadcast::Receiver<Operation>>,
+    pub limit: usize,
```
Comment (chubei) [inline dozer-api/src/grpc/common/service.rs:29]: Rename to `default_max_num_records `

File: dozer-api/src/grpc/typed/service.rs [hunk 3]
```diff
@@ -42,6 +42,7 @@ pub struct TypedService {
     endpoint_map: HashMap<String, TypedEndpoint>,
     event_notifier: Option<tokio::sync::broadcast::Receiver<Operation>>,
     security: Option<ApiSecurity>,
+    limit: usize,
```
Comment (chubei) [inline dozer-api/src/grpc/typed/service.rs:45]: Rename to `default_max_num_records`

File: dozer-api/src/rest/mod.rs [hunk 4]
```diff
@@ -48,6 +47,7 @@ pub struct ApiServer {
     cors: CorsOptions,
     security: Option<ApiSecurity>,
     host: String,
+    default_limit: usize,
```
Comment (chubei) [inline dozer-api/src/rest/mod.rs:50]: Rename to `default_max_num_records`

File: dozer-types/src/constants.rs [hunk 5]
```diff
@@ -5,3 +5,4 @@ pub const DEFAULT_CLOUD_TARGET_URL: &str = "https://api.dev.getdozer.io";
 pub const DEFAULT_QUERIES_DIRECTORY: &str = "queries";
 pub const DEFAULT_LAMBDAS_DIRECTORY: &str = "lambdas";
 pub const LOCK_FILE: &str = "dozer.lock";
+pub const DEFAULT_LIMIT: u32 = 50;
```
Comment (chubei) [inline dozer-types/src/constants.rs:8]:
```suggestion
pub const DEFAULT_LIMIT: usize = 50;
```

File: dozer-cli/src/simple/orchestrator.rs [hunk 6]
```diff
@@ -100,6 +101,7 @@ impl SimpleOrchestrator {
                 LmdbRwCacheManager::new(get_cache_manager_options(&self.config))
                     .map_err(OrchestrationError::CacheInitFailed)?,
             );
+            let limit = get_max_return_records_num(&self.config);
```
Comment (chubei) [inline dozer-cli/src/simple/orchestrator.rs:104]: Rename to `default_max_num_records`

File: dozer-cache/src/cache/expression/mod.rs [hunk 7]
```diff
@@ -51,7 +48,7 @@ impl QueryExpression {
 
 impl Default for QueryExpression {
     fn default() -> Self {
-        Self::with_default_limit()
+        Self::with_limit(DEFAULT_LIMIT.try_into().unwrap())
```
Comment (chubei) [inline dozer-cache/src/cache/expression/mod.rs:51]:
```suggestion
        Self::with_limit(DEFAULT_LIMIT)
```

File: dozer-types/src/models/config.rs [hunk 8]
```diff
@@ -84,6 +84,14 @@ pub struct Config {
     /// UDF specific configuration (eg. !Onnx)
     #[serde(default, skip_serializing_if = "Vec::is_empty")]
     pub udfs: Vec<UdfConfig>,
+
+    #[prost(uint32, tag = "16")]
+    #[serde(default = "default_limit_for_query")]
+    // max records to be returned from the endpoints
+    pub default_max_num_records: u32,
```
Comment (chubei) [inline dozer-types/src/models/config.rs:91]: Add it to `ApiConfig` instead of top level

File: dozer-types/src/models/config.rs [hunk 9]
```diff
@@ -84,6 +84,14 @@ pub struct Config {
...
+}
+pub fn default_limit_for_query() -> u32 {
```
Comment (chubei) [inline dozer-types/src/models/config.rs:93]:
```suggestion
pub fn default_default_max_num_records() -> u32 {
```

File: dozer-cli/src/utils.rs [hunk 10]
```diff
@@ -144,3 +144,7 @@ pub fn get_cache_manager_options(config: &Config) -> CacheManagerOptions {
         ..CacheManagerOptions::default()
     }
 }
+
+pub fn get_max_return_records_num(config: &Config) -> usize {
```
Comment (chubei) [inline dozer-cli/src/utils.rs:148]:
```suggestion
pub fn get_default_max_num_records(config: &Config) -> usize {
```

**Review by chubei — APPROVED** [review]

**Comment by chubei** [comment]
Thanks! Can you also make a doc PR? After the doc PR is merged, label this PR with "doc-update-completed".

**Comment by aaryaattrey** [comment]
Sure.

**Pull request merged**